datetime/src/main/java/com/fasterxml/jackson/datatype/jsr310/deser/InstantDeserializer.java

```diff
@@ -285,6 +288,20 @@ private ZoneId getZone(DeserializationContext context)
         return (_valueClass == Instant.class) ? null : context.getTimeZone().toZoneId();
     }
 
+    private String replaceZeroOffsetAsZIfNecessary(String text)
+    {
+        if (replaceZeroOffsetAsZ && endsWithZeroOffset(text)) {
```

Comment:
Thanks for the comment @cowtowncoder,  I just checked the doc, your point makes perfect sense, I was wrong, I remember if no match found that would throw an exception. will fix